src/main/scala/chisel3/aop/Select.scala

```diff
@@ -391,7 +394,9 @@ object Select {
     val isPort = module._component.get
       .asInstanceOf[DefModule]
       .ports
-      .flatMap { p => DataMirror.collectAllMembers(p.id) }
+      .map(_.id)
+      .collect { case (p: Data) => p }
+      .flatMap { p => DataMirror.collectAllMembers(p) }
       .contains(signal)
```

Comment:
In this situation, it actually seems fine to filter down to just Data, because the `signal` we are looking for in this API is just `Data`.